src/neptune/internal/utils/requirement_check.py

```diff
@@ -18,12 +18,14 @@
     "is_installed",
 ]
 
+from functools import lru_cache
 from importlib.util import find_spec
 from typing import Optional
 
 from neptune.exceptions import NeptuneMissingRequirementException
 
 
+@lru_cache
```

Comment:
If I remember we can provide an upper limit of how many values to store. Let's use something like `32` for instance.

Reply:
Thought of it too. Default is 128, but 32 should be enough too